sklearn/utils/estimator_checks.py

```diff
@@ -3050,6 +3055,126 @@ def check_class_weight_balanced_linear_classifier(name, Classifier):
     )
 
 
+@ignore_warnings(category=FutureWarning)
+def check_interaction_of_class_and_sample_weight_excluding_class(name, estimator_orig):
+    """Setting a class weights to zero is equivalent to excluding the samples
+    associated to this class from the calibration even when using non uniform
+    sample weights.
+    """
+    # Note: this test is similar to
+    # check_interaction_of_class_and_sample_weight_excluding_samples.
+    rng = np.random.RandomState(42)
+    err_msg_sw = (
+        f"For {name}, using class_weight as zero to one class is not "
+        "equivalent to making the sample_weight for the samples from that "
+        "class equals zero."
+    )
+    err_msg_exclude = (
+        f"For {name}, using class_weight as zero to one class is not "
+        "equivalent to excluding the samples from that class."
+    )
+
+    for n_classes in range(3, 10):
```

Comment:
Can you parametrize this test on `n_classes` using `pytest.mark.parametrize`?

Reply:
I'm afraid I don't know how to do it. I have added the decorator to the check functions, but they don't "propagate" to the test itself. These check functions are being yield by `_yield_all_checks` inside `parametrize_with_checks` where there is another `pytest.mark.parametrize` that must mess up with the previous one. I don't know...

```python
@pytest.mark.parametrize("n_classes", range(3, 10))
def check_interaction_of_class_and_sample_weight_excluding_class(
    name, estimator_orig, n_classes
):
...
```

Output from tests:
```python
 TypeError: check_interaction_of_class_and_sample_weight_excluding_class() missing 1 required positional argument: 'n_classes'
```

Is there another way to do what you asked?